third_party/move/move-compiler-v2/src/bytecode_generator.rs

```diff
@@ -826,6 +834,18 @@ impl<'env> Generator<'env> {
         })
     }
 
+    fn gen_deref(&mut self, targets: Vec<TempIndex>, id: NodeId, args: &[Exp]) {
+        // Optimize case of `Deref(Borrow(x)) => x`.
```

Comment:
Do we need to check whether x is copyable before doing this?

Reply:
Good observation. We get less copy errors with this optimization, and there is a baseline change accordingly in one of the tests. I think though this is fine, because the optimization is straightforward and predictable, and also ok on bytecode verification level, as the unnecessary copy we had before just doesn't appear there.